### DISCUSSION TITLE
How to cancel a  "table" editDisplayMode ? escape not working

### DISCUSSION THREAD

**Discussion by tttp** [KevinVandy/material-react-table#986] [Q&A]
Hi,

The documentation explains how to handle saving on the onBlur and it's clear, however, I don't understand how to cancel and edit

I was expecting that the escape key would do that, but it doesn't

Do I need to add  for each column a onKeyDown=(event) => {if (event.key === 'Escape') //skip the save in onBlur ?

is there an option I missed that does it automatically?

**Comment by KevinVandy** [comment]
nothing is stopping you from registering your own event listeners. Just set it up to call `table.setEditingRow(null)`

**Reply by tttp** [reply]
will do thanks, but am I right that out of the box it doesn't handle the esc key (it does handle the return one)

**Reply by tttp** [reply]
getting there, but I have one problem I couldn't resolve:
when pressing escape, I want to revert the modifications on the text field (not sure if better to use row.original or row._cacheValue) AND to exit the edit mode, ie. remove the edit field.

I tried table.setEditingCell(null), but it doesn't seem to do the trick.

@KevinVandy, what would be the way to cancel the edition?


```
   onKeyDown : (event) => {
      if (event.key === "Escape") {
        event.preventDefault();
        const id = props.column.id;
        event.target.value = props.row.original[id];
        props.table.setEditingCell(null);
      }
    }
```
